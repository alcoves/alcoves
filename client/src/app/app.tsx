import '@fontsource/inter'
import * as React from 'react'
import * as ReactDOM from 'react-dom/client'

import { theme } from '../config/theme.tsx'
import { router } from './routes.tsx'
import { ChakraProvider } from '@chakra-ui/react'
import { RouterProvider } from 'react-router-dom'
import { ColorModeScript } from '@chakra-ui/react'
import { AuthProvider } from '../context/authProvider.tsx'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <ColorModeScript initialColorMode={theme.initialColorMode} />
    <ChakraProvider theme={theme}>
      <QueryClientProvider client={queryClient}>
        <AuthProvider>
          <RouterProvider router={router} />
        </AuthProvider>
      </QueryClientProvider>
    </ChakraProvider>
  </React.StrictMode>
)
