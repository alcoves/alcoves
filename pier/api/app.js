require('dotenv').config();

const cors = require('cors');
const morgan = require('morgan');
const express = require('express');
const bodyParser = require('body-parser');
const responseTime = require('response-time');
const cookieParser = require('cookie-parser');

const app = express();

app.use(cors());
app.use(morgan('dev'));
app.use(responseTime());
app.use(cookieParser());
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: false }));

app.use('/', require('./routes/root'));
app.use('/me', require('./routes/me'));
app.use('/login', require('./routes/login'));
app.use('/videos', require('./routes/videos'));
app.use('/uploads', require('./routes/uploads'));
app.use('/register', require('./routes/register'));

app.use((req, res, next) => {
  const error = new Error('not found');
  error.status = 404;
  next(error);
});

app.use((error, req, res, next) => {
  res.status(500).send({
    message: error.message || 'unknown',
    error,
  });
});

module.exports = app;
