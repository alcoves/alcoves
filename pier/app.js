require('dotenv').config();

const cors = require('cors');
const morgan = require('morgan');
const express = require('express');
const bodyParser = require('body-parser');
const responseTime = require('response-time');
const cookieParser = require('cookie-parser');

const app = express();

app.use(cors());
app.use(morgan('dev'));
app.use(responseTime());
app.use(cookieParser());
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: false }));

app.use('/', require('./api/routes/root'));
// TODO :: Refactor into plural resource
app.use('/users', require('./api/routes/users'));
app.use('/posts', require('./api/routes/posts'));
app.use('/videos', require('./api/routes/videos'));
app.use('/uploads', require('./api/routes/uploads'));
app.use('/channels', require('./api/routes/channels'));

app.use((req, res, next) => {
  const error = new Error('not found');
  error.status = 404;
  next(error);
});

app.use((error, req, res, next) => {
  res.status(500).send({
    message: error.message || 'unknown',
    error,
  });
});

module.exports = app;
