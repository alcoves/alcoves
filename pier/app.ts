/* eslint-disable import/first */
require('dotenv').config();

import cors from 'cors'
import morgan from 'morgan'
import express from 'express'
import root from './routes/root'
import pods from './routes/pods'
import jobs from './routes/jobs'
import videos from './routes/videos'
import mongoose, { ConnectOptions } from 'mongoose';

if (!process.env.MONGODB_URI) throw new Error("MONGODB_URI must be defined!")
mongoose.connect(process.env.MONGODB_URI as string, {
  tls: true,
  useNewUrlParser: true,
  useUnifiedTopology: true,
  tlsCAFile: './ca-certificate.crt'
} as ConnectOptions);

const app = express();
app.use(cors())
app.use(express.json())
app.use(morgan('tiny'))

app.use('/', root)
app.use('/pods', pods)
app.use('/jobs', jobs)
app.use('/videos', videos)

export default app