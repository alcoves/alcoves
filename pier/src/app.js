const express = require('express');
const jwt = require('jsonwebtoken');
const { ApolloServer } = require('apollo-server-express');

const server = new ApolloServer({
  modules: [
    require('./gql/user'),
    require('./gql/video'),
    require('./gql/upload'),
  ],
  context: ({ req }) => {
    const token = req.headers.authorization || '';
    if (token) {
      if (token.includes('SERVICE')) {
        if (!token.split('SERVICE ')[1] === process.env.CONVERSION_API_KEY) {
          throw new Error('invalid service api key');
        }
      } else {
        const user = jwt.verify(token.split(' ')[1], process.env.JWT_KEY);
        return { user };
      }
    }
  },
  tracing: true,
});

const app = express();
server.applyMiddleware({ app });
module.exports = app;
