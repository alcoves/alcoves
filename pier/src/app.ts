/* eslint-disable import/first */
require('dotenv').config();

import express from 'express';
import root from './routes/root'
import videos from './routes/videos'

const app = express();

app.use('/', root)
app.use('/videos', videos)

export default app