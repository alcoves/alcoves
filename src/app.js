import React from 'react';

import Home from './components/Home';
import Dash from './components/Dash';
import Login from './components/Login';
import Video from './components/Video';
import Footer from './components/Footer';
import Account from './components/Account';
import Confirm from './components/Confirm';
import Register from './components/Register';
import Uploader from './components/Uploader';
import Editor from './components/Editor/Editor';
import Navigation from './components/Navigation';
import UserVideoGrid from './components/UserVideoGrid';
import ResetPassword from './components/ResetPassword';
import ResendActivationCode from './components/ResendActivationCode';

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Typography } from '@material-ui/core';

function NoMatch() {
  return <Typography variant='h2'>404</Typography>;
}

function Content(props) {
  return <div style={{ minHeight: 'calc(100vh - 100px)' }}>{props.children}</div>;
}

function AppRouter() {
  return (
    <Router>
      <Navigation />
      <Content>
        <Switch>
          <Route exact path='/' children={<Home />} />
          <Route exact path='/dash' children={<Dash />} />
          <Route exact path='/login' children={<Login />} />
          <Route exact path='/confirm' children={<Confirm />} />
          <Route exact path='/account' children={<Account />} />
          <Route exact path='/upload' children={<Uploader />} />
          <Route exact path='/videos/:id' children={<Video />} />
          <Route exact path='/register' children={<Register />} />
          <Route exact path='/editor/:id' children={<Editor />} />
          <Route exact path='/reset/password' children={<ResetPassword />} />
          <Route exact path='/users/:username' children={<UserVideoGrid />} />
          <Route exact path='/reset/code' children={<ResendActivationCode />} />
          <Route component={NoMatch} />
        </Switch>
      </Content>
      <Footer />
    </Router>
  );
}

export default function App({ children }) {
  return <AppRouter>{children}</AppRouter>;
}
