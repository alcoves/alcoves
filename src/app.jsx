import React from 'react';

import { Typography, } from '@material-ui/core';
import { Switch, Route, } from 'react-router-dom';
import Home from './components/Home';
import Video from './components/Video';
import Login from './components/Login';
import Footer from './components/Footer';
import Account from './components/Account';
import Confirm from './components/Confirm';
import Register from './components/Register';
import Uploader from './components/Uploader';
import Editor from './components/Editor/Editor';
import Navigation from './components/Navigation';
import SearchResults from './components/SearchResults';
import UserVideoGrid from './components/UserVideoGrid';

function NoMatch() {
  return <Typography variant='h2'>404</Typography>;
}

function Content({ children }) {
  return <div style={{ minHeight: 'calc(100vh - 100px)' }}>{children}</div>;
}

function AppRouter() {
  return (
    <div>
      <Navigation />
      <Content>
        <Switch>
          <Route exact path='/'><Home /></Route>
          <Route exact path='/search'><SearchResults /></Route>
          <Route exact path='/login'><Login /></Route>
          <Route exact path='/register'><Register /></Route>
          <Route exact path='/v/:id'><Video /></Route>
          <Route exact path='/confirm'><Confirm /></Route>
          <Route exact path='/account'><Account /></Route>
          <Route exact path='/upload'><Uploader /></Route>
          <Route exact path='/editor/:id'><Editor /></Route>
          <Route exact path='/u/:username'><UserVideoGrid /></Route>
          <Route component={NoMatch} />
        </Switch>
      </Content>
      <Footer />
    </div>
  );
}

export default function App({ children }) {
  return <AppRouter>{children}</AppRouter>;
}
