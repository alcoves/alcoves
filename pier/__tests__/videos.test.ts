import request from 'supertest'
import app from '../src/app'

describe("Videos Endpoint", () => {
  it('should respond 404', async () => {
    const res = await request(app)
      .get('/videos/id')
    expect(res.statusCode).toEqual(404)
  })

  it('should respond 200', async () => {
    const res = await request(app)
      .get('/videos/OYtQ4YBeTIeiH5voS7W_T')
    expect(res.statusCode).toEqual(200)
    expect(res.body).toMatchSnapshot()
  })
})