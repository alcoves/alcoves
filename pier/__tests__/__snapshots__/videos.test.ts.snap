// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos Endpoint should respond 200 1`] = `
Object {
  "created_at": "2021-09-09T02:43:58.127Z",
  "deleted_at": null,
  "duration": "134.6000000000000",
  "id": "OYtQ4YBeTIeiH5voS7W_T",
  "mpd_link": "https://cdn.bken.io/v/OYtQ4YBeTIeiH5voS7W_T/pkg/manifest.mpd",
  "percent_completed": "100",
  "status": "completed",
  "thumbnail": "https://cdn.bken.io/v/OYtQ4YBeTIeiH5voS7W_T/thumb.jpg",
  "title": "Supreme Court",
  "updated_at": "2021-09-09T02:43:58.127Z",
  "user_id": "ckrf3a06e000t01mg1oft9cxd",
  "views": 9,
  "visibility": "public",
}
`;
